packages/model-viewer/src/three-components/Renderer.ts

```diff
@@ -53,15 +55,33 @@ const $singleton = Symbol('singleton');
  * the texture.
  */
 export class Renderer extends EventDispatcher {
-  static[$singleton] = new Renderer({debug: isDebugMode()});
+  protected static[$singleton]: Renderer|null = null;
+  protected static[$singletonWithTransparency]: Renderer|null = null;
 
-  static get singleton() {
-    return this[$singleton];
+  static get singleton(): Renderer {
+    if (this[$singleton] == null) {
+      this[$singleton] = new Renderer({debug: isDebugMode()});
+    }
+    return this[$singleton] as Renderer;
+  }
+
+  static get singletonWithTransparency(): Renderer {
+    if (this[$singletonWithTransparency] == null) {
+        this[$singletonWithTransparency] = new Renderer({debug: isDebugMode(), isTransparent: true});
+    }
+    return this[$singletonWithTransparency] as Renderer;
   }
 
   static resetSingleton() {
-    this[$singleton].dispose();
+    if (this[$singleton] != null) {
+      (this[$singleton] as Renderer).dispose();
+    }
     this[$singleton] = new Renderer({debug: isDebugMode()});
```

Comment:
Should this be created only if the previous value was defined (aka inside the if statement)? Same for the singleton with transparency below.

Reply:
Good remark, thank you.